src/entities/vault.ts

```diff
@@ -0,0 +1,139 @@
+import { Address, BigDecimal, ethereum } from "@graphprotocol/graph-ts";
+import {
+  Vault,
+  VaultDeposit,
+  VaultUserPosition,
+  VaultWithdraw,
+} from "../../generated/schema";
+import { Vault as VaultABI } from "../../generated/VaultFactory/Vault";
+import {
+  BIG_DECIMAL_ZERO,
+  BIG_INT_ZERO,
+  VAULT_FACTORY_ADDRESS,
+} from "../constants";
+import { loadToken } from "./token";
+import { loadVaultStrategy } from "./vaultStrategy";
+
+export function createVault(vaultAddress: Address): Vault {
+  const vault = new Vault(vaultAddress.toHexString());
+  const vaultContract = VaultABI.bind(vaultAddress);
+
+  const vaultStrategy = loadVaultStrategy(
+    vaultContract.try_getStrategy().value
+  );
+
+  const tokenXCall = vaultContract.try_getTokenX();
+  const tokenYCall = vaultContract.try_getTokenY();
+  const tokenX = loadToken(tokenXCall.value);
+  const tokenY = loadToken(tokenYCall.value);
+
+  vault.factory = VAULT_FACTORY_ADDRESS.toHexString();
+  vault.strategy = vaultStrategy.id;
+  vault.lbPair = vaultContract.try_getPair().value.toHexString();
+  vault.tokenX = tokenX.id;
+  vault.tokenY = tokenY.id;
+
+  vault.totalBalanceX = BIG_DECIMAL_ZERO;
+  vault.totalBalanceY = BIG_DECIMAL_ZERO;
+  vault.totalValueLockedUSD = BIG_DECIMAL_ZERO;
+  vault.totalValueLockedAVAX = BIG_DECIMAL_ZERO;
+
+  vault.txCount = BIG_INT_ZERO;
+
+  vault.save();
+  return vault;
+}
+
+export function loadVault(id: Address): Vault {
+  const vault = Vault.load(id.toHexString());
+  if (!vault) {
+    return createVault(id);
+  }
```

Comment:
We should get rid of these lines, and make `loadVault ` return either `Vault` or `null`. A vault entity should only be created in the VaultFactory's`VaultCreated` event mapping. Otherwise, vaults not from our  `vault_factory_address` could potential be created, for example if someone were to fork our contracts